app/models/job_execution.rb

```diff
@@ -173,7 +173,7 @@ def setup!(dir)
       return false unless @repository.setup!(dir, @reference)
       commit = @repository.commit_from_ref(@reference, length: nil)
       tag = @repository.tag_from_ref(@reference)
-      @job.update_git_references!(commit: commit, tag: tag)
+      @job.update_git_references!(commit: commit, tag: tag) if commit || tag
```

Comment:
ok, invalid commit hash will not even get to this step, as it will fail at the `checkout` step https://github.com/zendesk/samson/blob/51fc8afd30746b3dfba730431da686da9c8fda0d/app/models/git_repository.rb#L24

I will undo this change.